snuba/snuba_migrations/replays/0006_add_is_archived_column.py

```diff
@@ -0,0 +1,112 @@
+from typing import List, Sequence, Tuple
+
+from snuba.clickhouse.columns import Column, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+new_columns: Sequence[Tuple[Column[Modifiers], str]] = [
+    (Column("is_archived", UInt(8, Modifiers(nullable=True))), "urls")
+]
+
+alter_forward: Sequence[Column[Modifiers]] = [
+    Column("platform", String(Modifiers(nullable=True)))
+]
+
+alter_backward: Sequence[Column[Modifiers]] = [
+    Column("platform", String(Modifiers(nullable=False, default="''")))
```

Comment:
Is the new column supposed to have a default? Have you confirmed that running the `MODIFY` with just the nullable removes the default = ''?

Reply:
@evanh If this is going to cause operational issues we can drop the nullability requirements for platform.

Reply:
It's not an operational concern, it's just that I'm not sure if the `default = ''` is preserved in your migration. If it is, you'll have a nullable column that is always setting a default of `''`. The default should appear when you do `DESCRIBE TABLE`. To test, I would see that it is there before the migration, run the migration, and then be sure it's gone.

Reply:
I'm going to drop the nullability requirements for platform.  If the need arises in the future, we can address it then.

Re: migration misunderstandings.  I'm looking at the SQL output and this ALTER statement makes sense to me.  Where am I going wrong?  If I'm altering a nullable column to not-null column it needs a default on the column, right?

> ALTER TABLE replays_local MODIFY COLUMN platform String DEFAULT '';

Reply:
Ah I see, yes OK this is fine. Thank you.